src/p99_bluesky/plans/fast_scan.py

```diff
@@ -0,0 +1,232 @@
+                )
+                step_counter += 1
+
+    yield from finalize_wrapper(
+        plan=inner_fast_scan_grid(
+            dets,
+            step_motor,
+            step_start,
+            step_end,
+            step_size,
+            scan_motor,
+            scan_start,
+            scan_end,
+            motor_speed,
+            snake_axes,
+        ),
+        final_plan=clean_up(),
+    )
+
+
+def _fast_scan_1d(
+    dets: list[AsyncReadable],
+    motor: Motor,
+    start: float,
+    end: float,
+    motor_speed: float | None = None,
+):
+    """
+    The logic for one axis fast scan, see fast_scan_1d and fast_scan_grid
+
+    In this scan:
+    1) The motor moves to the starting point.
+    2) The motor speed is changed
+    3) The motor is set in motion toward the end point
+    4) During this movement detectors are triggered and read out until
+      the endpoint is reached or stopped.
+    5) Clean up, reset motor speed.
+
+    Note: This is purely software triggering which result in variable accuracy.
+    However, fast scan does not require encoder and hardware setup and should
+    work for all motor. It is most frequently use for alignment and
+      slow motion measurements.
+
+    Parameters
+    ----------
+    detectors : list
+        list of 'readable' objects
+    motor : Motor (moveable, readable)
+
+    start: float
+        starting position.
+    end: float,
+        ending position
+
+    motor_speed: Optional[float] = None,
+        The speed of the motor during scan
+    """
+
+    # read the current speed and store it
+    old_speed = yield from bps.rd(motor.velocity)
+
+    def inner_fast_scan_1d(
+        dets: list[AsyncReadable],
+        motor: Motor,
+        start: float,
+        end: float,
+        motor_speed: float | None = None,
+    ):
+        LOGGER.info(f"Moving {motor.name} to start position = {start}.")
+        yield from bps.mv(motor, start)  # move to start
+
+        if motor_speed:
+            LOGGER.info(f"Set {motor.name} speed = {motor_speed}.")
+            yield from bps.abs_set(motor.velocity, motor_speed)
+
+        LOGGER.info(f"Set {motor.name} to end position({end}) and begin scan.")
+        yield from bps.abs_set(motor.user_setpoint, end)
+        # yield from bps.wait_for(motor.motor_done_move, False)
+        done = False
+
+        while not done:
+            yield from bps.trigger_and_read(dets + [motor])
+            done = yield from bps.rd(motor.motor_done_move)
+        yield from bps.checkpoint()
+
+    yield from finalize_wrapper(
+        plan=inner_fast_scan_1d(dets, motor, start, end, motor_speed),
+        final_plan=reset_speed(old_speed, motor),
+    )
+
+
+def check_within_limit(values: list, motor: Motor):
+    LOGGER.info(f"Check {motor.name} limits.")
+    lower_limit = yield from bps.rd(motor.low_limit_travel)
+    high_limit = yield from bps.rd(motor.high_limit_travel)
```

Comment:
perhaps move this to `p99_bluesky/src/stubs/check_motor_within_limit`?  with the view of moving into dodal later?